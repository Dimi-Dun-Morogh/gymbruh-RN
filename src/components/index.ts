export * from './common/Button';
export * from './common/IconButton';
export * from './exercise/ExerciseCreateForm';
export * from './common/Input';
export * from './exercise/ExerciseListItem';
export * from './common/TextBlock';
export * from './routine/RoutineListItem';
export * from './workout/WorkOutExerciseItem';
export * from './workout/WorkOutExercises';
export * from './workout/WorkOutSet';
export * from './workout/WorkOutSetInput';
export * from './workout/WorkOutSetHistoryItem';
