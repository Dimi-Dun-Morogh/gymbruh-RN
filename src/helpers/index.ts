export * from './generateId';
export * from './pieChartUtils';
export * from './soundPlayer';
