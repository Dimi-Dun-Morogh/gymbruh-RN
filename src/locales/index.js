import i18n from 'i18next';
import {initReactI18next} from 'react-i18next';
import {store} from '../redux/store';
import {getLocales} from 'react-native-localize';

const resources = {
  en: {
    translation: require('../locales/en.json'),
  },
  ru: {
    translation: require('../locales/ru.json'),
  },
};

export const defineLanguage = () => {
  const {languages, currentLanguage} = store.getState().appSettingsState;

  if (!currentLanguage.length) {
    const deviceLang = getLocales()[0].languageCode;
    if (languages.includes(deviceLang)) {
      store.dispatch({
        type: 'SET_LANGUAGE',
        payload: deviceLang,
      });
      return deviceLang;
    } else {
      store.dispatch({
        type: 'SET_LANGUAGE',
        payload: 'en',
      });
      return 'en';
    }
  }
  return currentLanguage;
};

i18n.use(initReactI18next).init({
  resources,
  lng: 'en',
  fallbackLng: 'en',
  interpolation: {
    escapeValue: false,
  },
});

export default i18n;
