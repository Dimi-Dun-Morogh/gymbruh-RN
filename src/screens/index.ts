export * from './bottomTabScreens/HomeScreen';
export * from './stackScreens/HistoryScreen';
export * from './bottomTabScreens/ExerciseScreen';
export * from './bottomTabScreens/RoutineScreen';
export * from './stackScreens/CreateExerciseScreen';
export * from './stackScreens/ExerciseDetailScreen';
export * from './stackScreens/CreateRoutineScreen';
export * from './stackScreens/ExerciseDetailScreen';
export * from './stackScreens/RoutineDetailsScreen';
export * from './stackScreens/WorkOutScreen';
