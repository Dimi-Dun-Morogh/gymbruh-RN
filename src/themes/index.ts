const dark = {
  textColorMain: '#fff',
  textColorSecond: 'yellow',
  textColorThird: '#01F814',
  bgcMain: '#8a63f2',
  bgcContent: '#2c1338',
  bgcSecondary: '#000',
  bgcTabs: '#000',
  bgcInput: '#1f222b',
  borderColorInput: '#8a63f2',
  selectedBorderColor: '#fff',
  selectedTextColor: '#fff',
};

export type Theme = typeof dark;

const light: Theme = {
  textColorMain: '#000',
  bgcMain: '#fff',
  bgcContent: '#fff5fe',
  bgcSecondary: '#FFF',
  bgcTabs: '#FFF',
  bgcInput: '#ebf0f7',
  borderColorInput: '#8a63f2',
  selectedBorderColor: '#fff',
  textColorSecond: 'blue',
  textColorThird: 'red',
  selectedTextColor: '#fff',
};

const themes = {
  dark,
  light,
};

export const themePicker = (isDark: boolean = true): Theme =>
  isDark ? themes.dark : themes.light;
