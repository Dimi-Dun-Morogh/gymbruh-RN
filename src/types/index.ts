import {InitialStateType as SettingsState} from '../redux/appSettings/appSettings.reducer';
import {InitialExerState} from '../redux/exercises/exercise.reducer';
import {INITIAL_STATE_TYPE as HistoryState} from '../redux/history/history.reducer';
import {InitialRoutineState as RoutinesState} from '../redux/routines/routine.reducer';

export type BackUpData = {
  settings: SettingsState;
  exercises: InitialExerState;
  history: HistoryState;
  routines: RoutinesState;
};
