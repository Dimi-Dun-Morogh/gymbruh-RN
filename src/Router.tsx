import React from 'react';
import {StyleSheet} from 'react-native';
import {NavigationContainer, useNavigation} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';

import {
  HistoryScreen,
  HomeScreen,
  ExercisesScreen,
  RoutineScreen,
  CreateExerciseScreen,
  CreateRoutineScreen,
  ExerciseDetailScreen,
  RoutineDetailsScreen,
  WorkOutScreen,
  SettingsScreen,
} from './screens';
import {
  RootStackScreenList,
  BottomTabScreenList,
  NavProp,
} from './types/routingTypes';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {IconButton as HeaderButton} from './components';
import {Theme} from './themes/';
import {useTheme} from './hooks/useTheme';
import {useTranslation} from 'react-i18next';
import i18n, {defineLanguage} from './locales/';
import {useAppSelector} from './hooks/storeHooks';
import {playASound} from './helpers';
const initi18n = i18n;

const Stack = createStackNavigator<RootStackScreenList>();

const Tab = createBottomTabNavigator<BottomTabScreenList>();

const Tabs = () => {
  const navigation = useNavigation<NavProp>();
  const [theme] = useTheme();

  const tabsAdditional = {
    tabBarActiveTintColor: '#fff',
    tabBarActiveBackgroundColor: '#8a63f2',
    tabBarInactiveTintColor: theme.textColorMain,
  };
  const soundOn = useAppSelector(state => state.appSettingsState.soundOn);
  const {t} = useTranslation();
  React.useEffect(() => {
    playASound.switchSound(soundOn);
    i18n.changeLanguage(defineLanguage());
  }, [soundOn]);

  return (
    <Tab.Navigator
      screenOptions={{
        ...styles(theme),
        ...tabsAdditional,
      }}
      sceneContainerStyle={{backgroundColor: theme.bgcContent}}>
      <Tab.Screen
        name="home"
        component={HomeScreen}
        options={{
          title: 'gymbruh',
          headerRight: () => {
            return (
              <HeaderButton
                iconName="settings"
                size={35}
                color={theme.textColorMain}
                onPress={() => navigation.navigate('settings')}
              />
            );
          },
          tabBarLabel: t('home'),
          tabBarIcon: ({color, size}) => (
            <Icon name="home" color={color} size={size} />
          ),
        }}
      />
      <Tab.Screen
        name="routine"
        component={RoutineScreen}
        options={{
          title: t('routines'),
          tabBarLabel: t('routines'),
          headerRight: () => {
            return (
              <HeaderButton
                iconName="add"
                size={35}
                color={theme.textColorMain}
                onPress={() => navigation.navigate('routineCreate')}
              />
            );
          },
          tabBarIcon: ({color, size}) => (
            <Icon name="repeat" color={color} size={size} />
          ),
        }}
      />
      <Tab.Screen
        name="exercises"
        component={ExercisesScreen}
        options={{
          title: t('exercises'),
          tabBarLabel: t('exercises'),
          headerRight: () => {
            return (
              <HeaderButton
                size={35}
                color={theme.textColorMain}
                iconName="add"
                onPress={() => navigation.navigate('exercCreate')}
              />
            );
          },
          tabBarIcon: ({color, size}) => (
            <Icon name="fitness-center" color={color} size={size} />
          ),
        }}
      />
    </Tab.Navigator>
  );
};

const StackNav = () => {
  const [theme] = useTheme();
  const {t} = useTranslation();
  return (
    <Stack.Navigator
      screenOptions={{
        cardStyle: {
          backgroundColor: theme.bgcContent,
        },

        headerTintColor: styles(theme).headerTitleStyle.color,
        ...styles(theme),
      }}>
      <Stack.Screen
        name="tabs"
        component={Tabs}
        options={{headerShown: false}}
      />
      <Stack.Screen
        name="history"
        options={{title: t('history')}}
        component={HistoryScreen}
      />
      <Stack.Screen
        name="exercCreate"
        component={CreateExerciseScreen}
        options={{
          title: t('create exercise'),
        }}
      />
      <Stack.Screen name="exercDetail" component={ExerciseDetailScreen} />
      <Stack.Screen
        name="routineCreate"
        component={CreateRoutineScreen}
        options={({route}) => ({
          title: route.params?.routine.name || t('create routine'),
        })}
      />
      <Stack.Screen
        name="routineDetails"
        component={RoutineDetailsScreen}
        options={({route, navigation}) => ({
          headerRight: ({tintColor}) => {
            return (
              <HeaderButton
                size={35}
                iconName="edit"
                color={tintColor}
                onPress={() =>
                  navigation.navigate('routineCreate', {
                    routine: route.params.routine,
                  })
                }
              />
            );
          },
        })}
      />
      <Stack.Screen
        name="workOut"
        options={{
          title: t('workout'),
        }}
        component={WorkOutScreen}
      />
      <Stack.Screen
        name="settings"
        options={{
          title: t('settings'),
        }}
        component={SettingsScreen}
      />
    </Stack.Navigator>
  );
};

const Router = () => {
  return (
    <NavigationContainer>
      <StackNav />
    </NavigationContainer>
  );
};

const styles = (theme: Theme) => {
  return StyleSheet.create({
    tabBarStyle: {
      backgroundColor: theme.bgcTabs,
      borderTopWidth: 0,
    },
    tabBarLabelStyle: {
      fontSize: 16,
      fontWeight: 'bold',
    },
    tabBarIconStyle: {
      color: theme.textColorMain,
    },
    headerStyle: {
      backgroundColor: theme.bgcMain,
    },
    headerTitleStyle: {
      color: theme.textColorMain,
    },
  });
};

export default Router;
