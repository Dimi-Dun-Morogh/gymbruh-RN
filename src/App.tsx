import React from 'react';
import Router from './Router';

import {Provider} from 'react-redux';
import {store} from './redux/store';
import {View} from 'react-native';

const App = () => {
  return (
    <Provider store={store}>
      <View style={{flex: 1, backgroundColor: '#2c1338'}}>
        <Router />
      </View>
    </Provider>
  );
};

export default App;
