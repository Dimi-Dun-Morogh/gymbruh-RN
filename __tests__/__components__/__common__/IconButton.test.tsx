import * as React from 'react';
import renderer from 'react-test-renderer';
import {IconButton} from '../../../src/components';

it('renders correctly', () => {
  const tree = renderer
    .create(<IconButton onPress={() => null} iconName="tune" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
