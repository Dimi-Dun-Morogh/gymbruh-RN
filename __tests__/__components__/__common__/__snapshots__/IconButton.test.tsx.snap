// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 30,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 30,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            Object {
              "marginRight": 0,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      Óê©
    </Text>
  </View>
</View>
`;
